### PR TITLE
Update single stac

### PR DESCRIPTION
**Pull request by matthewhanson**
**Related Issue(s):** #
#691 

**Proposed Changes:**

1. Make single-file STAC follow Catalog spec
2. Make single-file STAC a GeoJSON FeatureCollection with Items as features
3. Add collections array to single-file STAC

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec), and I have opened issue/PR #XXX to track the change.

### CODE REVIEW

File: extensions/single-file-stac/json-schema/schema.json [hunk 1]
```diff
@@ -5,6 +5,12 @@
   "description": "Single File STAC Extension to combine Collections and Items in single file catalog",
   "allOf": [
     {
+<<<<<<< HEAD
+      "$ref": "../../catalog-spec/json-schema/catalog.json"
+    },
+    {
+=======
+>>>>>>> 99aa7da2760462aac1d304af8af5c303f9287173
```
Comment (cholmes) [inline extensions/single-file-stac/json-schema/schema.json:13]: Looks like this is an unresolved merge conflict? 

Probably from my removal of item collection. I just threw in a feature collection reference, but I'm not sure if that's actually proper in json schema stuff.

**Comment by m-mohr** [comment]
@matthewhanson I updated the catalog schema in 7b29de2b8e8005c4a9174f5c3873a9290dd5b4fe, I think that was missing...

File: extensions/single-file-stac/README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 - **Scope: ItemCollection**
```
Comment (m-mohr) [inline extensions/single-file-stac/README.md:6]:
```suggestion
- **Scope: Catalog**
```

File: extensions/single-file-stac/README.md [hunk 3]
```diff
@@ -15,21 +15,13 @@ The Items in the single file catalog should not be merged with the Collection pr
...
-
-A Single File STAC is a GeoJSON FeatureCollection, with Features which are Items, but the Collections are still needed.
-
-| Field Name         | Type   | Description                                                  |
-| ------------------ | ------ | ------------------------------------------------------------ |
-| collections | [[Collection](../../collection-spec/collection-spec.md#collection-fields)] | An array of Collections that are used by any of the Items in the FeatureCollection. |
-| search | [Search Object](#search-object) | An optional field containing a search endpoint and parameters used to generate this Single File STAC |
-
-## Search Object
+A Single File STAC is a complete [Catalog Object](../../catalog-spec/catalog-spec.md) presented as a GeoJSON FeatureCollection contained in a single file
```
Comment (m-mohr) [inline extensions/single-file-stac/README.md:18]:
```suggestion
A Single File STAC is a complete [Catalog Object](../../catalog-spec/catalog-spec.md) presented as a [GeoJSON FeatureCollection](https://tools.ietf.org/html/rfc7946#section-3.3) contained in a single file.
```

File: extensions/single-file-stac/README.md [hunk 4]
```diff
@@ -15,21 +15,13 @@ The Items in the single file catalog should not be merged with the Collection pr
...
 
 | Field Name         | Type   | Description                                                  |
 | ------------------ | ------ | ------------------------------------------------------------ |
-| endpoint | string | The root endpoint of a STAC API used for this search.                  |
-| parameters | Map<string, Object> | A dictionary of all the parameters used for the search. |
+| type               | string | **REQUIRED.** Type of the GeoJSON Object. MUST be set to `FeatureCollection`. |
+| collections | [[Collection](../../collection-spec/collection-spec.md#collection-fields)] | An array of Collections that are used by any of the Items in the catalog. |
```
Comment (m-mohr) [inline extensions/single-file-stac/README.md:23]:
```suggestion
| collections | \[[Collection](../../collection-spec/collection-spec.md#collection-fields)] | An array of STAC Collections that are used by any of the Items in the catalog. |
```

File: extensions/single-file-stac/README.md [hunk 5]
```diff
@@ -15,21 +15,13 @@ The Items in the single file catalog should not be merged with the Collection pr
...
+| features    | [[Item](../../item-spec/item-spec.md#item-fields)] | An array of Items |
```
Comment (m-mohr) [inline extensions/single-file-stac/README.md:24]:
```suggestion
| features    | \[[Item](../../item-spec/item-spec.md#item-fields)] | An array of STAC Items |
```

File: extensions/single-file-stac/README.md [hunk 6]
```diff
@@ -15,21 +15,13 @@ The Items in the single file catalog should not be merged with the Collection pr
...
-## Search Object
+A Single File STAC is a complete [Catalog Object](../../catalog-spec/catalog-spec.md) presented as a [GeoJSON FeatureCollection](https://tools.ietf.org/html/rfc7946#section-3.3) contained in a single file.
 
 | Field Name         | Type   | Description                                                  |
 | ------------------ | ------ | ------------------------------------------------------------ |
-| endpoint | string | The root endpoint of a STAC API used for this search.                  |
-| parameters | Map<string, Object> | A dictionary of all the parameters used for the search. |
+| type               | string | **REQUIRED.** Type of the GeoJSON Object. MUST be set to `FeatureCollection`. |
+| collections | \[[Collection](../../collection-spec/collection-spec.md#collection-fields)] | An array of STAC Collections that are used by any of the Items in the catalog. |
+| features    | \[[Item](../../item-spec/item-spec.md#item-fields)] | An array of STAC Items |
```
Comment (m-mohr) [inline extensions/single-file-stac/README.md:24]:
```suggestion
| features    | **REQUIRED.** \[[Item](../../item-spec/item-spec.md#item-fields)] | An array of STAC Items |
```

**Comment by m-mohr** [comment]
I made a couple of changes (see the comments above) and fixed the JSON Schema. Should be fine to be merged now.

**Pull request merged**